src/main/scala/com/advancedtelematic/campaigner/client/DirectorClient.scala

```diff
@@ -37,10 +41,14 @@ class DirectorHttpClient(uri: Uri, httpClient: HttpRequest => Future[HttpRespons
   override def setMultiUpdateTarget(
     ns: Namespace,
     updateId: ExternalUpdateId,
-    devices: Seq[DeviceId]): Future[Seq[DeviceId]] = {
+    devices: Seq[DeviceId],
+    correlationId: CorrelationId): Future[Seq[DeviceId]] = {
     val path   = uri.path / "api" / "v1" / "admin" / "multi_target_updates" / updateId.value
     val entity = HttpEntity(ContentTypes.`application/json`, devices.asJson.noSpaces)
-    val req = HttpRequest(HttpMethods.PUT, uri.withPath(path), entity = entity).withNs(ns)
+    val req = HttpRequest(
+      HttpMethods.PUT,
+      uri.withPath(path).withQuery(Uri.Query(s"correlationId=${correlationId.value}")),
```

Comment:
Maybe easier to read?
`uri.withPath(path).withQuery(Query("correlationId" -> correlationId.value))`